src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -53,12 +53,32 @@ internal IArgument CreateArgumentOperation(ArgumentKind kind, IParameterSymbol p
 
         private IVariableDeclaration CreateVariableDeclarationInternal(BoundLocalDeclaration boundLocalDeclaration, SyntaxNode syntax)
         {
-            return OperationFactory.CreateVariableDeclaration(boundLocalDeclaration.LocalSymbol, Create(boundLocalDeclaration.InitializerOpt), _semanticModel, syntax);
+            IVariableInitializer initializer = null;
+            if (boundLocalDeclaration.InitializerOpt != null)
+            {
+                IOperation initializerValue = Create(boundLocalDeclaration.InitializerOpt);
+                SyntaxNode initializerSyntax;
+                bool isImplicit;
+                if (syntax is VariableDeclaratorSyntax variableDeclarator)
+                {
+                    initializerSyntax = variableDeclarator.Initializer;
+                    isImplicit = false;
+                }
+                else
+                {
+                    initializerSyntax = initializer.Syntax;
+                    isImplicit = true;
```

Comment:
When do we hit the implicit case?

Reply:
Basically whenever the syntax for the initializer and the initializer value are identical, we mark the initializer as implicit.

Reply:
Actually, there are no tests covering this path. Let me file an issue to track adding a test OR remove the else code.

Reply:
Filed https://github.com/dotnet/roslyn/issues/22574